src/meili/traffic_segment_matcher.cc

```diff
@@ -74,26 +74,29 @@ namespace {
       //get segments for this edge
       edge = marker.edge;
       const auto* directed_edge = tile->directededge(edge);
-      auto segments = tile->GetTrafficSegments(edge);
       //if there were no segments we'll start an invalid one to serve
       //as a placeholder for the section of the path that has no ots's
+      auto segments = tile->GetTrafficSegments(edge);
       if(segments.empty())
         segments = { valhalla::baldr::TrafficSegment{{}, marker.edge_distance, marker.edge_distance, true, true} };
       //the way id for this edge
       auto way_id = tile->edgeinfo(directed_edge->edgeinfo_offset()).wayid();
```

Comment:
the first part of this trick is to not store a `way_id` for transition edges as that makes no sense anyway